assets/js/amazon-wc-checkout.js

```diff
@@ -79,6 +79,51 @@
 			renderButton( '#pay_with_amazon_cart', 'cart' );
 		}
 
+		/* Handles Amazon Pay Button on Product Pages. */
+		if ( $( '#pay_with_amazon_product' ).length > 0 ) {
+			var amazonProductBtn = renderButton( '#pay_with_amazon_product', 'product' );
+			if ( null !== amazonProductBtn ) {
+				amazonProductBtn.onClick( function() {
+					if ( $( '.single_add_to_cart_button' ).hasClass( 'disabled' ) ) {
+						$( '.single_add_to_cart_button' ).trigger( 'click' );
+						return;
+					}
+					var elemToBlock = $( '#pay_with_amazon_product' ).closest( 'div.summary' ).length > 0 ? $( '#pay_with_amazon_product' ).closest( 'div.summary' ) : false;
+					if ( elemToBlock ) {
+						block( elemToBlock );
+					}
+					var pid = $( 'button[type="submit"][name="add-to-cart"]' ).attr( 'value' );
+					pid = pid || $( 'input[type="hidden"][name="product_id"]' ).val();
+					var qnt = $( 'input[type="number"][name="quantity"]' ).val();
+					var vid = $( 'input[type="hidden"][name="variation_id"]' ).length > 0 ? $( 'input[type="hidden"][name="variation_id"]' ).val() : false;
+					var qrs = '&pid=' + pid + '&qnt=' + qnt + ( vid ? '&vid=' + vid : '' );
+					$.ajax(
+						{
+							url: amazon_payments_advanced.ajax_url,
+							type: 'get',
+							data: 'action=' + amazon_payments_advanced.change_cart_action + '&_change_carts_nonce=' + amazon_payments_advanced.change_cart_ajax_nonce + qrs,
+							success: function( result ) {
+								if ( elemToBlock ) {
+									unblock( elemToBlock );
```

Comment:
It may be better to put boths calls to  `unblock()` in the `.always()` callback of `$.ajax`

Reply:
Done!